### PR TITLE
refactor: replace enzyme on RNTL

### PR DESCRIPTION
**Pull request by dimatretyak**
This PR resolves []

### Description
[Notion card](https://www.notion.so/artsy/remove-enzyme-from-eigen-b4efdc0fe1cd4f0682d3b4df9c73f628)

### Refactored tests
* BidResult
* AuctionPrice
* BidButton
* CommercialButtons
* ImageCarouselFullScreen
* ImageZoomView
* ImageCarouselContext
* ImageWithLoadingState
* PartnerCard
* FullFeaturedArtistList

### PR Checklist

- [ ] I tested my changes on **iOS** / **Android**.
- [ ] I added screenshots or videos to illustrate my changes.
- [ ] I added Tests and Stories for my changes.
- [ ] I added an [app state migration].
- [ ] I hid my changes behind a [feature flag].

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

### CODE REVIEW

File: src/app/Scenes/Artwork/Components/AuctionPrice.tsx [hunk 1]
```diff
@@ -82,9 +82,17 @@ export class AuctionPrice extends React.Component<AuctionPriceProps> {
             {!!myBidPresent && (
               <Text>
                 {myBidWinning ? (
-                  <CheckCircleIcon height="16" fill="green100" />
+                  <CheckCircleIcon
+                    height="16"
+                    fill="green100"
+                    accessibilityLabel="My Bid Winning Icon"
```
Comment (gkartalis) [inline src/app/Scenes/Artwork/Components/AuctionPrice.tsx:88]: 😍 👏 😍 👏 😍 👏

**Review by gkartalis — APPROVED** [review]
🚀 AMazing Job!!! 👏 😍 👏 😍 👏 😍 👏  Left some minor comments, let me know your thoughts!

File: src/app/Scenes/Collection/Components/FullFeaturedArtistList.tests.tsx [hunk 2]
```diff
@@ -1,88 +1,99 @@
...
+  })
+
+  it("renders without throwing an error", () => {
+    renderWithWrappersTL(<TestWrapper />)
 
-  it("renders without throwing an error", async () => {
-    expect(await render(FullFeaturedArtistListCollectionFixture))
+    resolveMostRecentRelayOperation(mockEnvironment, {
+      MarketingCollection: () => FullFeaturedArtistListCollectionFixture,
+    })
```
[the remark this answers is not in the document]
Comment (dimatretyak) [reply src/app/Scenes/Collection/Components/FullFeaturedArtistList.tests.tsx:46]: I think we can remove this test case

File: src/app/Components/Bidding/Screens/BidResult.tests.tsx [hunk 3]
```diff
@@ -1,98 +1,108 @@
...
-          bidderPositionResult={Statuses.winning}
-          sale_artwork={saleArtwork}
-          navigator={jest.fn() as any}
-        />
-      )
+      renderWithWrappersTL(<TestWrapper bidderPositionResult={Statuses.winning} />)
+
+      resolveMostRecentRelayOperation(mockEnvironment, {
+        Sale: () => sale,
+      })
```
[the remark this answers is not in the document]
Comment (dimatretyak) [reply src/app/Components/Bidding/Screens/BidResult.tests.tsx:81]: I think we can remove this test case

File: src/app/Scenes/Collection/Components/FullFeaturedArtistList.tests.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (gkartalis) [reply src/app/Scenes/Collection/Components/FullFeaturedArtistList.tests.tsx:46]: feel free to remove

**Review by gkartalis — APPROVED** [review]

File: src/app/Scenes/Artwork/Components/CommercialButtons/CommercialButtons.tsx [hunk 4]
```diff
@@ -13,6 +13,7 @@ export interface CommercialButtonProps {
   me: CommercialButtons_me$data
   // EditionSetID is passed down from the edition selected by the user
   editionSetID?: string
+  // FIXME: auctionState can be undefined
```
[the remark this answers is not in the document]
Comment (dimatretyak) [reply src/app/Scenes/Artwork/Components/CommercialButtons/CommercialButtons.tsx:16]: Added [a task](https://www.notion.so/artsy/Make-auctionState-prop-optional-for-the-CommercialButtons-component-8376532c07134bdead56698bbc588f13) to the Eigen Refining board

**Review by MounirDhahri — APPROVED** [review]
I can't thank you for this enough!

**Pull request merged**